eng/pipelines/templates/steps/build-test.yml

```diff
@@ -32,6 +33,11 @@ steps:
       pip freeze
     displayName: 'Prep Environment'
 
+  - ${{if eq(variables['System.TeamProject'], 'internal') }}:
```

Comment:
How does this work for local developers trying to build and test? Do they also need to enable authentication locally to be able to pull from this internal feed during install steps?

Reply:
They will need to add the authenticate feed to their `pip.conf` or `.pypirc`.

Reply:
I do have a possible alternative implementation. The way that the "auth" works is to create a connection of form

`https://build:***@pkgs.dev.azure.com/azure-sdk/internal/_packaging/azure-sdk-for-python-pr/pypi/simple`.

This is _exactly_ the same form as we'd provide as an argument to pip. Using another --extra-index-url.

There is definitely another way to attack this. I'm honestly fine with either, and was just trying to re-use the same methodology as already used for the publishing.

Reply:
After chatting with this I'm find with this approach.